app/population/middleware/set-population-id.js

```diff
@@ -0,0 +1,24 @@
+const locationsFreeSpacesService = require('../../../common/services/locations-free-spaces')
+
+async function setResultsPopulationTable(req, res, next) {
+  try {
+    const { locationId, date } = req.params
+
+    const freeSpaces = await locationsFreeSpacesService.getPrisonFreeSpaces({
+      dateFrom: date,
+      dateTo: date,
+      locationIds: locationId,
+    })
+
+    const population = freeSpaces?.[0]?.meta?.populations?.[0] || {}
```

Comment:
Looking at this, can a location have multiple populations? Are we sure the first item will always be the one we want?

Reply:
This request only retrieves one day worth of data, so the first item is definitely the one we want.